src/Factory/MatcherFactory.php

```diff
@@ -49,21 +49,23 @@ private function buildMatchers(Parser $parser, Backtrace $backtrace) : Matcher\C
 
     private function buildArrayMatcher(Matcher\ChainMatcher $scalarMatchers, Parser $parser, Backtrace $backtrace) : Matcher\ArrayMatcher
     {
-        $orMatcher = new Matcher\OrMatcher($backtrace, $scalarMatchers);
-
-        return new Matcher\ArrayMatcher(
+        $arrayMatcher = new Matcher\ArrayMatcher(
             new Matcher\ChainMatcher(
                 'array',
                 $backtrace,
                 [
-                    $orMatcher,
+                    new Matcher\OrMatcher($backtrace, $orMatchers = clone $scalarMatchers),
```

Comment:
have you tried it without a clone?

Reply:
Deleting the clone works but I added it here for a specific reason. Since the factory describes the following matcher priorities:

```
// Matchers are registered in order of matching
// 1) all scalars
// 2) json/xml
// 3) array
// 4) or "||"
// 5) full text
```

I have decided to clone the scalair matchers in order to keep theses priorities on the main matcher.

Reply:
and without clone that order is changed?